### PR TITLE
Use newer version of stack in CI

### PR DESCRIPTION
**Pull request by fkm3**
Required by #187.

The version we were using is old enough that it doesn't work with the
latest stackage LTS. haskellstack.org says

    There is also a Ubuntu package for Ubuntu 16.10 and up, but the
    distribution's Stack version lags behind, ...

So, instead of asking them to update it, it's probably better to
download the tar of the version we want.

Somehow updating stack surfaced a new pedantic warning in GradientTest,
so I've fixed that as well (and cleaned up some code using snake_case).

### CODE REVIEW

File: ci_build/Dockerfile [hunk 1]
```diff
@@ -12,8 +12,6 @@ ADD . /tfhs
 WORKDIR /tfhs
 
 RUN \
-    apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 575159689BEFB442 && \
```
Comment (blackgnezdo) [inline ci_build/Dockerfile]: This was slightly more end-to-end secure, but I guess we place ultimate trust into github anyway, so no big deal.

File: tensorflow-ops/tests/GradientTest.hs [hunk 2]
```diff
@@ -392,18 +392,18 @@ transAttrs a b =
 testConv2DBackpropInputGrad :: Test
 testConv2DBackpropInputGrad = testCase "testConv2DBackpropInputGrad" $ do
     (dx, shapeDX, shapeX) <- TF.runSession $ do
-        let conv_input_shape = TF.vector [1, 2, 2, 1 :: Int32] -- [batch, h, w, in_channels]
```
Comment (blackgnezdo) [inline tensorflow-ops/tests/GradientTest.hs]: Looks like an unrelated cleanup. Split it off?

File: ci_build/Dockerfile (hunk 1 above)
Comment (fkm3) [reply ci_build/Dockerfile]: Whoops, got trigger happy. Restored it.

File: tensorflow-ops/tests/GradientTest.hs (hunk 2 above)
Comment (fkm3) [reply tensorflow-ops/tests/GradientTest.hs]: Reverted.
